alchemy/src/neon/role.ts

```diff
@@ -0,0 +1,307 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import type { Secret } from "../secret.ts";
+import { handleApiError } from "./api-error.ts";
+import { createNeonApi, type NeonApiOptions, type NeonApi } from "./api.ts";
+import type { NeonProject } from "./project.ts";
+import type { NeonBranch } from "./branch.ts";
+
+/**
+ * Properties for creating or updating a Neon role
+ */
+export interface NeonRoleProps extends NeonApiOptions {
+  /**
+   * The project containing the branch
+   */
+  project: string | NeonProject;
+
+  /**
+   * The branch where the role will be created
+   */
+  branch: string | NeonBranch;
+
+  /**
+   * Name of the role
+   */
+  name: string;
+
+  /**
+   * Whether to adopt an existing role if it already exists
+   * @default false
+   */
+  adopt?: boolean;
+}
+
+/**
+ * A Neon database role with permissions
+ */
+export interface NeonRole
+  extends Resource<"neon::Role">,
+    Omit<NeonRoleProps, "apiKey"> {
+  /**
+   * The ID of the project containing this role
+   */
+  projectId: string;
+
+  /**
+   * The ID of the branch containing this role
+   */
+  branchId: string;
+
+  /**
+   * Name of the role
+   */
+  name: string;
+
+  /**
+   * Password for the role (if available)
+   */
+  password?: Secret;
+
+  /**
+   * Whether this role is protected from deletion
+   */
+  protected: boolean;
+
+  /**
+   * Time at which the role was created
+   */
+  createdAt: string;
+
+  /**
+   * Time at which the role was last updated
+   */
+  updatedAt: string;
+}
+
+/**
+ * Creates a Neon database role with permissions.
+ *
+ * @example
+ * // Create a basic role for application access:
+ * const role = await NeonRole("app-user", {
+ *   project: project,
+ *   branch: branch,
+ *   name: "app_user"
+ * });
+ *
+ * @example
+ * // Create a role for read-only access:
+ * const role = await NeonRole("readonly-user", {
+ *   project: project,
+ *   branch: branch,
+ *   name: "readonly_user"
+ * });
+ *
+ * @example
+ * // Adopt an existing role if it already exists:
+ * const role = await NeonRole("existing-role", {
+ *   project: project,
+ *   branch: branch,
+ *   name: "legacy_user",
+ *   adopt: true
+ * });
+ */
+export const NeonRole = Resource(
```

Comment:
I noticed project.ts has a NeonRole, can we make sure all NeonRole code and types are here in this file and imported into project.ts